main.tf

```diff
@@ -63,6 +63,13 @@ resource "azurerm_role_assignment" "attach_reader_role" {
   role_definition_name = "Reader"
 }
 
+# Give the service principal a Storage Blob Data Reader role in the Subscription
+resource "azurerm_role_assignment" "storage_blob_data_reader_role" {
```

Comment:
can you add `count = var.set_tenant_level_permissions == true ? 1 : 0`
like below one and also we need at more place 
https://github.com/uptycslabs/terraform-azurerm-tenant-integration/blob/ce60c48bf7c201b116afed04f38a774ca6b81aed/main.tf#L104

Reply:
Ok